src/Engine/GameEngine.cs

```diff
@@ -81,5 +88,55 @@ public void OnError(Exception error)
         {
             throw new NotImplementedException();
         }
+
+        #region IDisposable Implementation
+
+        private bool _disposedValue = false;
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposedValue)
+            {
```

Comment:
I can change this, that's not a problem, I did originally do this but for whatever reason I obviously changed it.